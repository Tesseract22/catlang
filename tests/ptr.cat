proc main() {
    let x := 5;
    let ref := x.&;

    ref.* = 10;

    print(ref);
    print(x);

    let ref_ref := ref.&;
    print(ref_ref);

    let y := 69;
    ref_ref.* = y.&;

    print(ref.*);

    let sum := ref.* + ref_ref.*.*;
    print(sum);
}