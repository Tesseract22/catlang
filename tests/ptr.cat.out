0x7fffd5708348
10
0x7fffd5708340
69
138
